src/Languages/Editor/Impl/ContainedLanguage/ContainedLanguageHost.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information
+
+using Microsoft.Languages.Editor.Composition;
+using Microsoft.Languages.Editor.Services;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Languages.Editor.ContainedLanguage {
+    /// <summary>
+    /// Helper class that simplifies location of contained language host
+    /// for contained (secondary) languages. 
+    /// </summary>
+    public static class ContainedLanguageHost {
```

Comment:
Not really but this class simplifies things quite a bit. IContainedLanguageHostProvider is exported per content type in ordered fashion so code that imports proper one is error-prone. This class takes just view and buffer and hides ordering and content type acrobatics.